### PR TITLE
[SPARK-30057][DOCS]Add a statement of platforms Spark runs on

### PR DESCRIPTION
**Pull request by huangtianhua**

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
Just 2 cents as I'm not a committer nor a PMC member of Apache Spark;

I think there's still a non-trivial gap between "all UTs pass" and "guaranteed to work". Even Spark runs around 3~4hrs of UTs for every PRs as well as master branch, the benefit of UTs is more alike avoiding regressions and it doesn't prevent any further bugs including correctness and dataloss.

IMHO, the core circumstances of "guaranteed to work" are following:

1) Have end-to-end system tests running in such environment as well. (If it includes chaos monkey then much better.) I believe some of vendors are running their own system tests in their own environments.
2) Core staffs of project are familiar with such environment and able to deal with any env. specific issues.
3) End users have been running various workloads so far. I agree that's egg vs chicken problem for ARM support, but at least we may want to see couple of references for (pseudo) real workloads.

but everyone may have different opinions on the circumstances. Someone may have even higher bar on this, someone may have lower bar - so please just treat this as 2 cents, as I mentioned above.

**Comment by huangtianhua** [comment]
recheck

**Comment by huangtianhua** [comment]
retest this please

File: docs/index.md [hunk 1]
```diff
@@ -42,9 +42,7 @@ If you'd like to build Spark from
 source, visit [Building Spark](building-spark.html).
 
 
-Spark runs on both Windows and UNIX-like systems (e.g. Linux, Mac OS). It's easy to run
-locally on one machine --- all you need is to have `java` installed on your system `PATH`,
-or the `JAVA_HOME` environment variable pointing to a Java installation.
+Spark runs on both Windows and UNIX-like systems (e.g. Linux, Mac OS), and it should run on any platform that runs a supported version of Java, this should include JVMs on x86_64 and ARM64. It's easy to run locally on one machine --- all you need is to have `java` installed on your system `PATH`, or the `JAVA_HOME` environment variable pointing to a Java installation.
```
Comment (srowen) [inline docs/index.md]: "of Java, this should" -> "of Java. This should"

**Review by srowen — CHANGES_REQUESTED** [review]
@HeartSaVioR yes that's true, but I don't see any stronger claim like that being made here?

**Comment by HeartSaVioR** [comment]
That depends on how individual interprets these sentences. For me, I don't just interpret "run on" as "can be launched", but interpret as "guaranteed to execute properly", which implicitly denotes there's community support.

That leads me to the questions: have we try to configure "cluster" with ARM machines and run some workloads what we normally do? Which workloads we've checked with manual tests/system tests? If there's an issue which doesn't occur on X86_64, who in community can investigate on it?

**Comment by srowen** [comment]
If true, it's as true of x86_64, and as true of just about any big piece of software, right? I don't think we can define "guaranteed to execute properly" let alone guarantee it. To that end, I don't think a stronger statement is possible or necessary.

I don't think silence implies x86_64. The platform Spark runs on is almost entirely the JVM.

Your question is really, is there anything like the same amount of testing on ARM? Well, @huangtianhua has been running the tests on ARM, yes. This can continue externally, and maybe one day be a part of the amplab cluster. If I'm not worried about it, it's just that I doubt there is much reason to believe Spark will break regularly on JVM-on-ARM vs JVM-on-x86_64. 

All that said, I am also kind of neutral on being explicit here. We've already pushed back on official announcements about 'support' for essentially the reasons you advance. This statement seemed like a minor compromise. If it helps buy some good will and thus contribution in the form of ARM testing, fixes? OK.

File: docs/index.md (hunk 1 above)
Comment (huangtianhua) [reply docs/index.md]: Sure, will correct this.

**Comment by huangtianhua** [comment]
@HeartSaVioR @srowen Thanks for your attention of this. Yes, we have integrated the maven and python tests into amplab jenkins CI, the arm tests as daily jobs and have been stablely running for few weeks. And as @srowen said, the platform Spark runs on is almost entirely the JVM, I hope to add this to Spark docs to make users know Spark should run on any platform that runs a supported version of Java, it should include ARM, not X86_64 only. Also this will make users have more interest, confidence and will in running Spark on ARM platform.

**Comment by zhengruifeng** [comment]
@srowen +1

I am neutral on this PR, but it seem reasonable to expose to some degree the efforts made to support various platforms.
And I found that the commuity investigated on cross-platform issues like [using StrictMath.log instead of math.log](https://github.com/apache/spark/pull/25279)

**Comment by HeartSaVioR** [comment]
> The platform Spark runs on is almost entirely the JVM.

"leveldb-jni" thing is one of counter-examples of this. Spark relies on some libraries which leverage JNI. I'm seeing now only LevelDB, but RocksDB would be another one if we integrate into Spark. (Ecosystem already has RocksDB implementation of StateStore which would narrow the support range of platforms.)

What I wanted to say is, explicitly documenting about platforms, architectures, etc. is nearly same effect on announcing, and even longer impact as saying we are officially supporting it - so I feel it's not just trivial doc change. But as I commented earlier, all my comments are just 2 cents and have no impact if community (maybe this means the consensus among PMC members most of times) doesn't have same voice. If we think Spark community has consensus of adding this on doc, that should be good to go.

**Comment by srowen** [comment]
Yes, 'almost entirely'. Compressions codecs and the optional BLAS library are other examples. Agree with the issue about giving more assurance that it's supported than there is testing, but this minimal statement mostly communicates that it doesn't not-work in general.

**Comment by HyukjinKwon** [comment]
Fwiw a lot of functionalities dont work too on Windows but we declare Spark supports it.

**Comment by clr-apache** [comment]
I am not a PMC member nor committer but I do understand the efforts that have been made to integrate ARM CI platform testing on Spark. This testing seems to support the statement that Spark "should run" on "X86_64 and ARM64".
Including such a statement would seem to have a positive effect on increasing the potential users of Spark, which is consistent with our corporate mission of providing software for the public good.

**Comment by srowen** [comment]
Merged to master

**Comment by cloud-fan** [comment]
late LGTM

**Pull request closed without merging**